section2/exercises/if-statements.js

```diff
@@ -31,19 +31,25 @@ if (weather === 'sunny') {
   console.log('good to go!');
 }
 
-/* 
+/*
 YOU DO:
-Using the dogAge variable defined below, 
-determine if a dog is a puppy (2 or younger), 
+Using the dogAge variable defined below,
+determine if a dog is a puppy (2 or younger),
 an adult, or elderly (10 or older).
 
 Log to the console the appropriate age range
 (puppy, adult, elderly).
 */
 
-var dogAge = 3;
+var dogAge = 1;
 // Write your conditional here
-
+if (dogAge <= 2) {
+  console.log('Puppy');
+} else if (dogAge <= 9) {
+  console.log('Adult');
+} else if (dogAge >= 10) {
+  console.log('Elderly');
+}
```

Comment:
What if the `dogAge = 9.5`? Try running your program with this value and see what happens and report back here. This is a good example of missing edge cases in your code and it doing unexpected things.

We always should strive for _elegant_ code (ie simple and straightforward with as little lines as possible). A more elegant way to write this would be:
```
if (dogAge <= 2 && dogAge >= 0) {
  console.log('Puppy');
} else if (dogAge < 10) {
  console.log('Adult');
} else if (dogAge >= 10) {
  console.log('Elderly');
} else {
  console.log('Invalid dogAge'); // this is another example of an edge case where the dogAge could equal -1
}
```